include/envoy/thread/thread.h

```diff
@@ -2,40 +2,115 @@
 
 #include "envoy/common/pure.h"
 
+#include "absl/base/thread_annotations.h"
+
 namespace Envoy {
 namespace Thread {
 
 /**
- * Like the C++11 "basic lockable concept" but a concrete interface vs. a template.
+ * Like the C++11 "basic lockable concept" but a pure virtual interface vs. a template, and
+ * with thread annotations.
  */
-class BasicLockable {
+class LOCKABLE BasicLockable {
 public:
   virtual ~BasicLockable() {}
 
-  virtual void lock() PURE;
-  virtual bool try_lock() PURE;
-  virtual void unlock() PURE;
+  virtual void lock() EXCLUSIVE_LOCK_FUNCTION() PURE;
+  virtual bool tryLock() EXCLUSIVE_TRYLOCK_FUNCTION(true) PURE;
```

Comment:
IMO this is an Envoy interface, not a std:: interface, so it should follow Envoy conventions, hence the change from try_lock to tryLock.

Reply:
FYI this was done previously so it worked with std::unique_lock. If we are no longer using std::unique_lock than no problem to change.